vendor/assets/javascripts/behaviors/selectable_asset.js

```diff
@@ -10,6 +10,9 @@ Cmsify.SelectableAsset.prototype.select = function() {
   $preview.html(this.renderAssetOriginalPreview());
   $preview.append($('<p />').html(this.getFileName()));
   $('.js-remove-asset').removeClass('uk-hidden');
+  if (this.$el.data('asset-name') === 'download') {
+    $('#resource_item_name').val(this.getFileName());
```

Comment:
Need to select a generic name field, something like `$([id$=_name])` so it can work for other item types in the future, not just resources.